test/geometry/adjoints.jl

```diff
@@ -53,25 +53,56 @@ end
     end # TODO: Remove the @test_broken after tests work.
 end
 
-@testset "SVector{N, T} gradient" begin
+@testset "SVector{N, T} and SMatrix{N, N, T, L} gradient" begin
     for t in (Float32, Float64, RGB{Float32}, RGB{Float64})
         inp = rand(t, 2)
+        inp_mat = rand(t, 3, 3)
         _sep(x) = x.r + x.g + x.b
         if t ∈ (Float32, Float64)
             @test Zygote.gradient(x->sum(SVector{2, t}(x)), inp)[1] == ones(t, 2)
+            @test Zygote.gradient(x->sum(SMatrix{3, 3, t, 9}(x)), inp_mat)[1] == ones(t, 3, 3)
         else
             @test Zygote.gradient(x->_sep(sum(SVector{2, t}(x))), inp)[1] == ones(t, 2)
+            @test Zygote.gradient(x->_sep(sum(SMatrix{3, 3, t, 9}(x))), inp_mat)[1] == ones(t, 3, 3)
```

Comment:
The tests in both the branches of the if blocks seems the same modulo `_sep`, should we get rid of the if block perhaps

Reply:
This seems like a good idea. I'll make this consistent by placing an `abs` equivalent.